relayer/cli/src/commands/query/connection.rs

```diff
@@ -83,24 +90,26 @@ impl Runnable for QueryConnectionEndCmd {
 
         let chain = TendermintChain::from_config(chain_config).unwrap();
         // run with proof:
-        // cargo run --bin relayer -- -c simple_config.toml query connection end ibc0 ibconeconnection
+        // cargo run --bin relayer -- -c relayer/relay/tests/config/fixtures/simple_config.toml query connection end ibc-test connectionidone --height 3
         //
         // run without proof:
-        // cargo run --bin relayer -- -c simple_config.toml query connection end ibc0 ibconeconnection -p false
+        // cargo run --bin relayer -- -c relayer/relay/tests/config/fixtures/simple_config.toml query connection end ibc-test connectionidone --height 3  -p false
         //
         // Note: currently both fail in amino_unmarshal_binary_length_prefixed().
         // To test this start a Gaia node and configure a client using the go relayer.
-        let res = block_on(query_connection(
+        let res = block_on(query::<TendermintChain, ProtoConnectionEnd, ConnectionEnd>(
             &chain,
-            opts.height,
-            opts.connection_id.clone(),
-            opts.proof,
+            Request {
+                path: Some(TendermintPath::from_str(&"store/ibc/key").unwrap()),
+                data: ConnectionPath::new(opts.connection_id)
+                    .to_string()
+                    .into_bytes(),
+                height: Some(Height::from(opts.height)),
+                prove: opts.proof,
+            },
         ));
 
-        match res {
-            Ok(cs) => status_info!("connection query result: ", "{:?}", cs.connection),
-            Err(e) => status_info!("connection query error", "{}", e),
-        }
+        println!("{:?}", res);
```

Comment:
The previous code with the `match` was slightly better, because it did not expose the whole stacktrace in case of errors, and printed the specific `cs.connection` object details.

```suggestion
        match res {
            Ok(cs) => status_info!("connection query result: ", "{:?}", cs.connection),
            Err(e) => status_info!("connection query error", "{}", e),
        }
```

Reply:
I thought that thing is just for debugging. The CLI should return a better-formed result. But sure, the result has to be unpacked. I'll submit a commit for it. (and for the same in channel.)

(Sidenote: Please note that the type of `res` is different than the earlier `cs`. `res` is already the data, there's no need to call `.connection`.)

Reply:
I thought that thing is just for debugging. The CLI should return a better-formed result. But sure, the result has to be unpacked. I'll submit a commit for it. (and for the same in channel.)

(Sidenote: Please note that the type of res is different than the earlier cs. res is already the data, there's no need to call .connection.)